src/main/java/org/folio/rest/transaction/PaymentCreditHandler.java

```diff
@@ -0,0 +1,387 @@
+      log.info("Calculating budget totals for payments/credits with no Encumbrances attached");
+      noEncumbranceBudgetsGrouped.entrySet()
+        .forEach(this::updateBudgetNoEncumbranceTotals);
+    }
+
+    return groupedBudgets;
+
+  }
+
+  private Budget updateBudgetPaymentTotals(Map.Entry<Budget, List<Transaction>> entry) {
+    Budget budget = entry.getKey();
+    CurrencyUnit currency = Monetary.getCurrency(entry.getValue()
+      .get(0)
+      .getCurrency());
+    entry.getValue()
+      .stream()
+      .forEach(txn -> {
+        budget.setExpenditures(MoneyUtils.sumMoney(budget.getExpenditures(), txn.getAmount(), currency));
+        budget.setAwaitingPayment(MoneyUtils.subtractMoney(budget.getAwaitingPayment(), txn.getAmount(), currency));
+      });
+
+    return budget;
+  }
+
+  private Budget updateBudgetCreditTotals(Map.Entry<Budget, List<Transaction>> entry) {
+    Budget budget = entry.getKey();
+    CurrencyUnit currency = Monetary.getCurrency(entry.getValue()
+      .get(0)
+      .getCurrency());
+    entry.getValue()
+      .stream()
+      .forEach(txn -> {
+        budget.setExpenditures(MoneyUtils.subtractMoneyNonNegative(budget.getExpenditures(), txn.getAmount(), currency));
+        budget.setEncumbered(MoneyUtils.sumMoney(budget.getEncumbered(), txn.getAmount(), currency));
+      });
+    return budget;
+  }
+
+  /**
+   * For the payments and credits that do not have corresponding encumbrances, the available and unavailable amounts in the budget
+   * are re-calculated
+   *
+   * @param entry- Budget with list of transactions that do have encumbrances
+   */
+  private Budget updateBudgetNoEncumbranceTotals(Map.Entry<Budget, List<Transaction>> entry) {
+    Budget budget = entry.getKey();
+    CurrencyUnit currency = Monetary.getCurrency(entry.getValue()
+      .get(0)
+      .getCurrency());
+    entry.getValue()
+      .stream()
+      .forEach(txn -> {
+        if (isCreditTransaction.test(txn)) {
+          budget.setAvailable(MoneyUtils.sumMoney(budget.getAvailable(), txn.getAmount(), currency));
+          budget.setUnavailable(MoneyUtils.subtractMoney(budget.getUnavailable(), txn.getAmount(), currency));
+        } else if (isPaymentTransaction.test(txn)) {
+          budget.setAvailable(MoneyUtils.subtractMoney(budget.getAvailable(), txn.getAmount(), currency));
+          budget.setUnavailable(MoneyUtils.sumMoney(budget.getUnavailable(), txn.getAmount(), currency));
+        }
+
+      });
+    return budget;
+  }
+
+  /**
+   * Update the Encumbrance transaction attached to the payment/Credit(from paymentEncumbranceID)
+   * in a transaction
+   *
+   * @param tx : the list of payments and credits
+   */
+  private Future<Tx<List<Transaction>>> updateEncumbranceTotals(Tx<List<Transaction>> tx) {
+    Boolean noEncumbrances = tx.getEntity()
+      .stream()
+      .allMatch(tx1 -> StringUtils.isBlank(tx1.getPaymentEncumbranceId()));
+
+    if (Boolean.TRUE.equals(noEncumbrances)) {
```

Comment:
`allMatch` returns primitive boolean, so `noEncumbrance` can be primitive as well and '==' can be used instead of equals